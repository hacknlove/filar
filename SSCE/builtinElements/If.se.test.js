const { DOMParser } = require("linkedom");
const { processAllElements } = require("../processAllElements");
const { customElementsMap } = require("../common");
const parser = new DOMParser();

const If = require("./If.se");

describe("If", () => {
  it("renders the content if the condition is truthy", async () => {
    const document = parser.parseFromString(`
            <html>
                <body Condition="{|true|}">
                    <If key="Condition">
                        <span>content</span>
                    </If>
                </body>
            </html>
        `);

    customElementsMap.set("If", If);

    await processAllElements(document);

    expect(document.toString()).toMatchSnapshot();
  });
  it("does not render the content if the condition is falsy", async () => {
    const document = parser.parseFromString(`
            <html>
                <body Condition="{|false|}">
                    <If key="Condition">
                        <span>content</span>
                    </If>
                </body>
            </html>
        `);

    customElementsMap.set("If", If);

    await processAllElements(document);

    expect(document.toString()).toMatchSnapshot();
  });
  it("renders the content if the condition is truthy (expression)", async () => {
    const document = parser.parseFromString(`
            <html>
                <body>
                    <If key="Condition" Condition="{|5 > 4|}">
                        <span>content</span>
                    </If>
                </body>
            </html>
        `);

    customElementsMap.set("If", If);

    await processAllElements(document);

    expect(document.toString()).toMatchSnapshot();
  });

  it("does not render the content if the condition is falsy (expression)", async () => {
    const document = parser.parseFromString(`
            <html>
                <body>
                    <If key="Condition" Condition="{|5 < 4|}">
                        <span>content</span>
                    </If>
                </body>
            </html>
        `);

    customElementsMap.set("If", If);

    await processAllElements(document);

    expect(document.toString()).toMatchSnapshot();
  });
});
