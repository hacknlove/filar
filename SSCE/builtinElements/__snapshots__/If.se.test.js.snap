// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If does not render the content if the condition is falsy (expression) 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
            <html>
                <body>
                    
                </body>
            </html>
        "
`;

exports[`If does not render the content if the condition is falsy 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
            <html>
                <body>
                    
                </body>
            </html>
        "
`;

exports[`If renders the content if the condition is truthy (expression) 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
            <html>
                <body>
                    
                        <span>content</span>
                    
                </body>
            </html>
        "
`;

exports[`If renders the content if the condition is truthy 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
            <html>
                <body>
                    
                        <span>content</span>
                    
                </body>
            </html>
        "
`;
